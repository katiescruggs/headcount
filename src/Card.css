ul.card-data {
    list-style-type: none;
}

.hide {
  visibility: hidden;
}