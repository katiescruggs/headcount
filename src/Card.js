import React from 'react';

const Card = ({districtName, districtData}) => (
  <div className="card">
    <div className="card-hed">
      <h2>{districtName}</h2>
    </div>
    <div className="card-data">
      
      {
     
      
      }
     

    </div>
  </div>
)

export default Card