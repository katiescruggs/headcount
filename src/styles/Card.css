h2 {
    display: block;
    -webkit-margin-before: 0;
    -webkit-margin-after: 0;
    -webkit-margin-start: 0px;
    -webkit-margin-end: 0px;
}

.card-head {
    display: flex;
    justify-content: center;
    background: #26517e;
    color: white;
    font-size: 15px;
    text-align: center;
    height: 75px;
    align-items: center;
}
.card-head h2 {
  font-size: 20px;
  margin: 0 20px;
}

.card {
  box-shadow: 1.5px 3px 3px 1.5px #566782;
  background: #ffffff;
  margin: 10px;
  width: 230px;
}

.card-body {
  padding: 20px;
}

.card-clicked .card-head {
  background-color: #cb252d;
}

.card-clicked .compare-button{
  background-color: #cb252d;
}

.card-clicked .compare-button:hover{
  background-color: #ffffff;
  border: 1px solid #cb252d;
}

.button {
    display: flex;
    justify-content: center;
}

.compare-button {
  width: 70%;
  background: #26517e;
  color: #ffffff;
  height: 30px;
}

.compare-button:hover{
  color: #26517e;
  background: #ffffff;
  border: 1px solid #26517e;
}

ul.card-data {
  list-style-type: none;
  padding: 0;
  margin: 0 0 10px 35px;
}


.icon-span {
  width: 20px;
  display: inline-block;
  color: #139fdc;
}
