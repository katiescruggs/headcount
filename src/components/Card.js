import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Card.css';
import Check from 'react-icons/lib/fa/check';


const checkIcon = (num) => {
  return num >= 0.5 ? <Check /> : null;
};

const displayData = (districtData) => {
  const years = Object.keys(districtData);

  return years.map((year, index) => {
    return (
      <li key={index}> 
        <span className="icon-span">
          {checkIcon(districtData[year])} 
        </span> 
        <strong>{year}</strong>
        : {districtData[year]}
      </li>
    );
  });
};

const buttonClick = (handleClick, districtName) => {
  handleClick(districtName);
};

const Card = ({districtName, districtData, handleClick, type, buttonText}) => {
  if (districtData !== undefined) {
    return (
      <div className={`card ${type}`}>
        <div className="card-head">
          <h2>{districtName}</h2>
        </div>
        <div className = "data-list">
          <ul className="card-data">
            {displayData(districtData)}
          </ul>
        </div>
        <div className = "button">
          <button 
            className = "compare-button" 
            onClick={ () => buttonClick(handleClick, districtName)}>
            {buttonText}
          </button>
        </div>
      </div>
    );
  } else {
    return null;
  }
};

export default Card;

Card.propTypes = {
  districtName: PropTypes.string,
  districtData: PropTypes.object
};