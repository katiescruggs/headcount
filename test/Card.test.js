import React from 'react'
import { shallow } from 'enzyme'
import Card from '../src/components/Card.js'

describe ('Card Tests', () => {
  it('has a card class name, an h2, an ul, and a button', () => {
    const district = {location: 'Academy 20', data: {2014: 1, 2015: 1} }
    const mockHandleClick = jest.fn()
    const wrapper = shallow ( <Card 
            key={district.location}
            id={district.location}
            districtName={district.location}
            districtData={district.data}
            handleClick = {mockHandleClick}
            type= ''
            buttonText='Compare' /> );
    expect(wrapper.hasClass('card')).toEqual(true);
    expect(wrapper.find('button').length).toEqual(1);
    expect(wrapper.find('ul').length).toEqual(1);
    expect(wrapper.find('h2').length).toEqual(1);
  });

  it('should display li elements based on how much data is passed in', () => {
    const mockHandleClick = jest.fn()
    const district1 = {location: 'Academy 20', data: {2014: 1, 2015: 1} }
    const wrapper1 = shallow ( <Card 
            key={district1.location}
            id={district1.location}
            districtName={district1.location}
            districtData={district1.data}
            handleClick = {mockHandleClick}
            type= ''
            buttonText='Compare' /> );

    const district2 = {location: 'Colorado', data: {2014: 0.3, 2015: 1, 2016: 0.6} }
    const wrapper2 = shallow ( <Card 
            key={district2.location}
            id={district2.location}
            districtName={district2.location}
            districtData={district2.data}
            handleClick = {mockHandleClick}
            type= ''
            buttonText='Compare' /> );

    expect(wrapper1.find('li').length).toEqual(2);
    expect(wrapper2.find('li').length).toEqual(3);
  });

  it('should render a checkmark next to the data if the value is greater than 0.5', () =>{
    const district = {location: 'Colorado', data: {2014: 0.3, 2015: 1, 2016: 0.6} }
    const mockHandleClick = jest.fn()
    const wrapper = shallow ( <Card 
            key={district.location}
            id={district.location}
            districtName={district.location}
            districtData={district.data}
            handleClick = {mockHandleClick}
            type= ''
            buttonText='Compare' /> );
    expect(wrapper.find('FaCheck').length).toEqual(2);
  });

  it('should call the handleClick function if the compare button is clicked', () => {
    const district = {location: 'Colorado', data: {2014: 0.3, 2015: 1, 2016: 0.6} }
    const mockHandleClick = jest.fn()
    const wrapper = shallow ( <Card 
            key={district.location}
            id={district.location}
            districtName={district.location}
            districtData={district.data}
            handleClick = {mockHandleClick}
            type= ''
            buttonText='Compare' /> );

    expect(mockHandleClick.mock.calls.length).toEqual(0);

    wrapper.find('button').simulate('click');
    expect(mockHandleClick.mock.calls.length).toEqual(1);
  });
});