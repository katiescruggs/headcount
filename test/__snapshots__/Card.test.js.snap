exports[`Card Tests should match the snapshot 1`] = `
<div
  className="card  Academy 20">
  <div
    className="card-head">
    <h2>
      Academy 20
    </h2>
  </div>
  <div
    className="card-body">
    <ul
      className="card-data">
      <li
        key="0">
        <span
          className="icon-span">
          <FaCheck />
        </span>
        <strong>
          2014
        </strong>
        : 
        1
      </li>
      <li
        key="1">
        <span
          className="icon-span" />
        <strong>
          2015
        </strong>
        : 
        0.3
      </li>
    </ul>
    <div
      className="button">
      <button
        className="compare-button"
        onClick={[Function]}>
        Compare
      </button>
    </div>
  </div>
</div>
`;
